In [2]:
import pandas as pd
from bokeh.charts import BoxPlot, Scatter, show, output_notebook

auto = pd.read_csv("Automobile_price_data _clean.csv")
autoND = pd.read_csv("Automobile_price_data _NoDash.csv")

In [3]:
auto.head()

make fuel-type aspiration num-of-doors body-style drive-wheels  \
0      honda       gas        std          two  hatchback          fwd   
1  chevrolet       gas        std          two  hatchback          fwd   
2     nissan    diesel        std          two      sedan          fwd   
3     toyota    diesel        std         four  hatchback          fwd   
4     toyota       gas        std         four      sedan          fwd   

  engine-location  wheel-base  length  width  ...    engine-size  fuel-system  \
0           front        86.6   144.6   63.9  ...             92         1bbl   
1           front        88.4   141.1   60.3  ...             61         2bbl   
2           front        94.5   165.3   63.8  ...            103          idi   
3           front        95.7   166.3   64.4  ...            110          idi   
4           front        95.7   166.3   64.4  ...             98         2bbl   

   bore stroke  compression-ratio horsepower peak-rpm city-mpg  highway-mpg  \
0  2.91   3.41                9.6         58     4800       49           54   
1  2.91   3.03                9.5         48     5100       47           53   
2  2.99   3.47               21.9         55     4800       45           50   
3  3.27   3.35               22.5         56     4500       38           47   
4  3.19   3.03                9.0         70     4800       38           47   

   price  
0   6479  
1   5151  
2   7099  
3   7788  
4   7738  

[5 rows x 24 columns]

In [4]:
autoND.head()

make fueltype aspiration numofdoors  bodystyle drivewheels  \
0      honda      gas        std        two  hatchback         fwd   
1  chevrolet      gas        std        two  hatchback         fwd   
2     nissan   diesel        std        two      sedan         fwd   
3     toyota   diesel        std       four  hatchback         fwd   
4     toyota      gas        std       four      sedan         fwd   

  enginelocation  wheelbase  length  width  ...    enginesize  fuelsystem  \
0          front       86.6   144.6   63.9  ...            92        1bbl   
1          front       88.4   141.1   60.3  ...            61        2bbl   
2          front       94.5   165.3   63.8  ...           103         idi   
3          front       95.7   166.3   64.4  ...           110         idi   
4          front       95.7   166.3   64.4  ...            98        2bbl   

   bore stroke  compressionratio horsepower peakrpm citympg  highwaympg  price  
0  2.91   3.41               9.6         58    4800      49          54   6479  
1  2.91   3.03               9.5         48    5100      47          53   5151  
2  2.99   3.47              21.9         55    4800      45          50   7099  
3  3.27   3.35              22.5         56    4500      38          47   7788  
4  3.19   3.03               9.0         70    4800      38          47   7738  

[5 rows x 24 columns]

In [3]:
p = BoxPlot(auto, values = "price", label = ["make", "body-style"], color = "make", plot_width = 800, plot_height = 600, title = "Price Summary (grouped by make,body-style)")
output_notebook()
show(p)

Loading BokehJS ...

The above line is supposed to produce a graph but GitHub is not able to show it. These differ from seaborn and matplotlib graphs because they are interactive. 
You can view the graph here: http://htmlpreview.github.io/?https://github.com/MrRoper/JNDataSci/blob/master/Lab10%20First%20Graph.html

In [4]:
tooltips=[
  ('Price', '@price'),
  ('Make', '@make'),
  ('Body-style', '@bodystyle'),
  ('City-mpg', '@citympg'),    
]

p2 = Scatter(autoND, x = "citympg", y = "price", color = "make", marker = "bodystyle", title = "LAS350A - Lab 10: Price vs City-mpg (shaded by Make, maker by Body-style)", plot_width = 800, plot_height = 600, tooltips=tooltips)
output_notebook()
show(p2)

Loading BokehJS ...

Same thing as the first one. The only difference is this one has tooltips.
You can view the graph here: http://htmlpreview.github.io/?https://github.com/MrRoper/JNDataSci/blob/master/Lab10%20Second%20Graph.html

The most efficient city-mpg sedan is a Nissan. It has a city mpg of 45 and a price of 7099.